wordpress-mobile/WordPress-Android
# [Jetpack Social] Implement Twitter deprecation notice container on Details screen

Fixes #18431 and #18429

To test the publicize details UI:
1 - Install JP app and log in;
2 - Select a site that has Twitter connected;
3 - Open Sharing screen;
4 - Observe the services list: Twitter should not be listed with the other services, but Twitter deprecation container should be shown below;
5 - Tap the Twitter row: publicize details screen should be opened, and Twitter deprecation notice should be shown there too. `Disconnect` button should be shown.
6 - Repeat the same with WP app.

To test the analytics events:
1 - Install JP app and log in;
2 - Select a site that has Twitter connected;
3 - Open Sharing screen;
4 - Tap the `Find out more` button: the event `twitter_notice_link_tapped` should be tracked with the property `{"source":"social_connection_list"}`
<img width="874" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/cf96aeae-5124-45c2-8f79-bf32cbf304c1">


5 - Go back to the publicize list screen and tap the Twitter row;
6 - Tap the `Find out more` button in the publicize details screen: the event `twitter_notice_link_tapped` should be tracked with the property `{"source":"social_connection_detail"}`
<img width="879" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/4fba7af8-c97c-4922-a2d6-7a53be27edee">

## Regression Notes
1. Potential unintended areas of impact
Publicize details screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Implemented `PublicizeTwitterDeprecationNoticeAnalyticsTrackerTest.kt`

PR submission checklist:


UI Changes testing checklist:
